### ISSUE TITLE
BUG - 2 - Android - Accessibility: Screen reader does not read error message content on form properly

### ISSUE THREAD

**Issue by SKMOstudio** [department-of-veterans-affairs/va-mobile-app#11349]
## What happened?

When a screen reader user on Android generates an error message when submitting evidence for a claim form\, the screen reader is not properly reading the content an order of the error messages properly.

## Specs:

- Device: Samsung Galaxy S24
- OS: Android
- User Account: Demo
- Accessibility State: Screen reader ON, default settings.

## Steps to Reproduce

1. Log into the VA app as a Demo user. Ensure the screen reader is ON.
2. Go to the Claims page.
3. Tap on an active Claim, then tap the `Submit evidence` CTA button.
4. Go through the process to submit evidence.
5. On the Upload files page, delete the selected file to upload. Then, tap the `Submit file` CTA button.
6. Observe - The screen reader will jump to focus on the error message generated upon submitting the blank evidence form, but will not focus on the header of the error message box.

## Desired behavior

1. When the error message box appears, the screen reader should jump focus to the header line of the error message box  "We need more information".
2. As the user navigates line-by-line through the error message box, the screen reader should read out the context in the correct order:
2.1 - "We need more information."
2.2 - "To submit this file, provide this information:"
2.3 - [Bullet list of missing files, document selection, or check box marked error].

## Acceptance Criteria

When a screen reader user submits a file for evidence, if there is an error in the form, the screen reader should jump focus to the top of the error message and contain context as to the errors that have occurred within the form.

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: **High** Low 
- Frequency: High **Low** 

- 3 - Low
- 2 - **High**
- 1 - Critical

## Screen shot(s) and additional information

https://github.com/user-attachments/assets/77cf862e-6bf6-46dc-868b-526f30356287

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, global, Health, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the relevant team's bug epic ([Global](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8302), [Health & Benefits](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8303), [API](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8304), [QART](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8305))

**Comment by matthew-df** [comment]
@TimRoe Hi Tim! Is this something someone from your team can look into? I mentioned above the issues we are seeing and possible steps for resolution. Let me know if you have any questions for me!

**Comment by TimRoe** [comment]
Hello @matthew-df, 
cc: @caw310 for visibility

If you look at the details of the bug I'd made #11863 or [the associated slack thread prior to creating](https://dsva.slack.com/archives/C018V2JCWRJ/p1761334373982409), there's [links to a PR](https://github.com/department-of-veterans-affairs/va-mobile-library/pull/654) that has an Android video showing the functionality behaves as expected in the component itself (mobile component storybook testing environment). I suspect the issue is caused by the app-level wrapper for the Alert component setting `role='alert'`.

Since the component works as intended in our testing environment on Android, I don't think this bug should need component changes to resolve.

That said, two notes:
1. I do not recall offhand why the component itself didn't include a role of 'alert' when created, but it may have been related to not wanting to conflate with "alert" dialogues for the mobile form factor which automatically include "alert" to the screen reader I believe--this should potentially be revisited
   - At face value, I do not think this is a bug and more of a general enhancement--one that would require design consideration because based on [the description of the alert role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/alert_role) it should only apply to error and maybe warning alerts, not info/success so there's some nuance around potentially incorporating
2. It is true that the component doesn't account for scroll to behavior and maybe needs ability to set the focus inside the component--this was a gray area when the component was created and expected the mobile core team would've come back at implementation time for updates if it wasn't working as intended and no issue was ever raised so no further action was taken
    - There could be a component bug here if the scroll to behavior can't be made to work after resolving the general bugged behavior in VAHB (as demonstrated with my bug report), but I'd like to see evidence it's definitely a problem after fixing the app-level issue before making component changes for the scroll to case
    - Even if the scroll to behavior after fixing the VAHB issues with Android Alerts works, it may have some behavior that could be improved by adding the ability to send a `ref` as you mention so there may be a non-bug, general enhancement option with this one too

**Comment by matthew-df** [comment]
@TimRoe Thanks for that info! I had a similar thought on the wrapper. I checked the alert without the wrapper and even went so far as to only render the alert in the app! But I saw the same screen reader interactions. I just checked the storybook with expo and see that it is correctly setting the accessibility points.  Ill take another look at this to see what else could possibly be affecting the component.

**Comment by SKMOstudio** [comment]
Accepted

Based on re-reviewing other pages with errors forms, the current behavior is consistent, and does _not_ require reading out "Error" as part of the context clue for the user.
